src/interfaces.ts

```diff
@@ -271,6 +283,7 @@ export interface ViewPortPlan {
 
   prevComponent?: ViewPortComponent;
   prevModuleId?: string;
+  prevViewModel?: string | Function | null;
```

Comment:
In which way is a `viewModel` of type `string` different from `moduleId`?

Reply:
This could be changed. The interface is just a merge of `moduleId` and `viewModel`, as I was not sure how to pass things around nicely. Mostly to do something like this:
```js
if (typeof prevViewModel === 'string' || prevViewModel === null) {
  // it was moduleId
} else {
  // it was viewModel
}
```

Reply:
null could also come from `viewModel` factory:
```js
viewModel: () => null
```
But I think we can treat it like `moduleId: null`